src/promises/__tests__/allNoLeaks.unit.test.ts

```diff
@@ -1,14 +1,12 @@
+import { setTimeout } from 'node:timers/promises';
 import { fail } from 'node:assert';
 import { allNoLeaks } from '../allNoLeaks';
 
 describe('allNoLeaks', () => {
 	it('should throw an error after companion long-duration `Promise` completes', async () => {
 		let longTaskCompleted = false;
-		const longTask = new Promise<void>((resolve) => {
-			setTimeout(() => {
-				longTaskCompleted = true;
-				resolve();
-			}, 50);
+		const longTask = setTimeout(50).then(() => {
```

Comment:
I wonder if this is for sure the same functionality. I can try it out to see. Well, yeah, the unit test I wrote here still works correctly, duh. :tada: